src/safeposix/syscalls/fs_calls.rs

```diff
@@ -5083,59 +5083,127 @@ impl Cage {
+                    // Since semaphores are consistent across all cages and all addresses within the cages
                     for offset in segment.semaphor_offsets.iter() {
-                        let sementry = cage2.sem_table.get(&(addrs[0] + *offset)).unwrap().clone(); //add  semaphors into semtable at addr + offsets
+                        let sementry = cage2.sem_table.get(&(addrs[0] + *offset)).unwrap().clone(); 
                         self.sem_table.insert(shmaddr as u32 + *offset, sementry);
                     }
                 }
             }
-
+            
+            // Map the shared segment onto the current cage using `map_shm` function
             segment.map_shm(shmaddr, prot, self.cageid)
         } else {
             syscall_error(Errno::EINVAL, "shmat", "Invalid shmid value")
         }
     }
 
-    //------------------SHMDT SYSCALL------------------
-
+    ///------------------SHMDT SYSCALL------------------
+    /// ### Description 
+    /// 
+    /// This syscall can be viewed as a reversal of the `shmat_syscall`
+    /// 'shmat_syscall` attaches a particular memory shared memory segment to a 
+    /// `shmaddr` passed as an argument
+    /// `shmdt` unmaps the shared memory segment that is currently mapped at the address
+    /// specified by the `shmaddr` argument. 
+    /// 
+    /// ### Arguments
+    /// 
+    /// `shmaddr` : The address within the address space of the calling Cage to unmap
+    /// 
+    /// ### Returns
+    /// 
+    /// Returns the id of the memory segment that has been unmapped
+    /// 
+    /// ### Errors
+    /// 
+    /// This function can result in the following errors
+    /// 
+    /// * EINVAL : If there is no memory segment with the address specified
+    /// 
+    /// 
+    /// ### Panics
+    /// 
+    /// This function panics if creating an inode fails
+    /// 
+    /// For more information please refer - [https://man7.org/linux/man-pages/man3/shmdt.3p.html]
     pub fn shmdt_syscall(&self, shmaddr: *mut u8) -> i32 {
         let metadata = &SHM_METADATA;
         let mut rm = false;
+        // Acquire the lock of the reverse memory mappings of the Cage object
         let mut rev_shm = self.rev_shm.lock();
+        // This function returns the index where the pair of (shmaddr, shmid) is store within the vector 
+        // That holds these mappings
         let rev_shm_index = Self::rev_shm_find_index_by_addr(&rev_shm, shmaddr as u32);
 
+        // Check if the index is valid
         if let Some(index) = rev_shm_index {
+            // Get the second element of the (shmaddr,shmid) pair which gives us the id of the memory segment
             let shmid = rev_shm[index].1;
+            // Get the memory segment from the shmtable which corresponds to the shmid extracted above
             match metadata.shmtable.entry(shmid) {
+                // If the memory segment is occupied
```

Comment:
Can you explain, what are we doing in the below section, when the memory segment is occupied?